### PR TITLE
Separate estimation code, shrink create_model

### PR DESCRIPTION
**Pull request by karlb**
More small steps on the way to make the data flow more clear.

The first commit is just moving code around, the real changes are in the smaller commits.

### CODE REVIEW

File: causing/estimate.py [hunk 1]
```diff
@@ -526,23 +526,31 @@ def estimate_biases(model_dat):
...
+    xmean = model_dat["xdat"].mean(axis=1)
+    xcdat = model_dat["xdat"] - xmean.reshape(model_dat["mdim"], 1)
+    ymmean = model_dat["ymdat"].mean(axis=1)
+    ymcdat = model_dat["ymdat"] - ymmean.reshape(model_dat["pdim"], 1)
+
     selvec = zeros(model_dat["ndim"])
     selvec[[list(model_dat["yvars"]).index(el) for el in model_dat["ymvars"]]] = 1
     selmat = diag(selvec)
     selvec = diag(selmat)
     selvec = diag(selmat)
```
Comment (Holger-Bartel) [inline causing/estimate.py]: this line is redundant and selvec is already known from two lines before

**Review by Holger-Bartel — APPROVED** [review]
Nice, a lot of smaller steps into the right direction. 
See the one comment, still.

File: causing/estimate.py (hunk 1 above)
Comment (karlb) [reply causing/estimate.py]: This is exactly what happened before in the code, it was just spread out a bit more. But I'm happy to remove it.

Comment (Holger-Bartel) [reply causing/estimate.py]: Yeah, I know it already happened before, but now it became quite obvious that it could be improved upon :)

**Pull request merged**